src/app/components-link.directive.ts

```diff
@@ -0,0 +1,39 @@
+import { Directive, ElementRef, Input, OnDestroy, OnInit } from '@angular/core';
+import { combineLatest, Subject } from 'rxjs';
+import { map, takeUntil } from 'rxjs/operators';
+import { ComponentsListSearchService } from './components-list.service';
+
+@Directive({
+  selector: '[npgComponentLink]',
+})
+export class ComponentLinkDirective implements OnInit, OnDestroy {
+  constructor(private el: ElementRef, private componentsListSearchService: ComponentsListSearchService) {}
```

Comment:
Order class members in a following way:
private members
public undecorated
public decorated
constructor
lifecycle methods
own methods